src/Dumplie/Application/Query/Customer/Result/Cart.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare (strict_types = 1);
+
+namespace Dumplie\Application\Query\Customer\Result;
+
+use Dumplie\Application\Exception\InvalidArgumentException;
+
+final class Cart
+{
+    /**
+     * @var CartItem[]|array
+     */
+    private $items;
+
+    /**
+     * @var float
+     */
+    private $totalPrice;
+
+    /**
+     * @param array|CartItem[] $items
+     * @throws InvalidArgumentException
+     */
+    public function __construct(array $items = [])
+    {
+        $this->totalPrice = 0;
+
+        foreach ($items as $item) {
+            if (!$item instanceof CartItem) {
+                throw new InvalidArgumentException();
+            }
+
+            $this->totalPrice += $item->price();
+        }
+
+        $this->items = $items;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEmpty() : bool
+    {
+        return !count($this->items);
+    }
+
+    /**
+     * @return float
+     */
+    public function totalPrice() : float
+    {
+        return $this->totalPrice;
+    }
+
+    /**
+     * @return string
+     */
+    public function currency() : string
+    {
+        return $this->isEmpty() ? null : $this->items[0]->currency();
```

Comment:
Przy takim sprawdzaniu currency, jak jest tu, pojawią się dwa problemy. Bo skoro założyliśmy, że nasz sklep operuje 1 walutą, to ja mając firmę Krzew, która sprzedawała na rynku europejskim dajmy na to w Euro, mając dobrze prosperujący biznes chcę zrobić ekspansje na nowy rynek gdzie walutą podstawową będzie dolar, jeśli uzależniamy nasz produkt od waluty, to teraz będę musiał dla swoich 10 tyś. produktów ją zmieniać. Dlatego, wydaje mi się lepiej jest operować na samej wartości, a o jednostce powinien wiedzieć sam system, ale nie koniecznie sam produkt. 

A drugi case to kiedy sprawdzamy pierwszy element, to możemy mieć przypadek iż dodaliśmy do koszyka produkt, potem dorzuciliśmy jeszcze 5, po czym stwierdzamy a ten 1 nam się nie podoba, usuwamy, w liście zostają elementy 1-4, a my sprawdzamy walutę w pierwszym, a trochę bez sensu by było zmieniać referencji do elementów w tablicy, dla takiego przypadku, żeby to nam ładnie wykrywało